composer/callbacks/checkpoint_saver.py

```diff
@@ -287,6 +290,9 @@ def __init__(
         num_checkpoints_to_keep: int = -1,
         weights_only: bool = False,
         ignore_keys: Optional[Union[list[str], Callable[[dict], None]]] = None,
+        save_folder: str = '',
```

Comment:
let's say the save_folder is "s3://bucket/path". The folder is just "path", because originally the rud was constructed out side of this class. But with this PR we need to build the backend inside this class, so have to pass in the full remote path with the backend name.